### PR TITLE
Feature/server/197 server start activitydiagrams of patients if execution is activated

### PR DESCRIPTION
**Pull request by YIllmann**
This PR connects the patient activity diagrams with the state changes of the game master. Therefore the web_api method is extended by the required logic. Further every runtime entity, that is processed during the call, contains a method to pass the command to the activity diagram. 

The implementation of the switch to FINISHED is skipped and only marked as TODO, because it requires a more complex archive structure, which should be done in a separate issue.

### CODE REVIEW

File: server/execution/entities/execution.py [hunk 1]
```diff
@@ -28,6 +30,28 @@ def __init__(self, id: int, name: str, scenario: Scenario,
         self.starting_time = starting_time
         self.notifications = []
 
+    def start_execution(self):
+        """ Performs the execution start protocol. """
+        if not self.scenario:
+            raise InternalServerError("Execution without a scenario detected.")
```
Comment (simohlsen) [inline server/execution/entities/execution.py:36]: In general, who should be responsible for having valid scenarios/executions? The scenario admin interface?

File: server/execution/entities/stategraphs/activity_diagram.py [hunk 2]
```diff
@@ -2,42 +2,66 @@
...
+        with self.lock.acquire_timeout(timeout=ACQUIRE_TIMEOUT) as acquired:
+            if not acquired:
+                raise TimeoutError
+            if self.current.is_state_outdated():
+                self.__update_state(self.current)
+
+            try:
+                new_state_uuid = self.current.treatments[treatment]
+                self.current = self.__get_state_by_id(new_state_uuid)
+                if self.current.timelimit != -1:
```
Comment (simohlsen) [inline server/execution/entities/stategraphs/activity_diagram.py:60]: Maybe `>= 0` ?

File: server/execution/entities/stategraphs/patientstate.py [hunk 3]
```diff
@@ -17,6 +17,9 @@ def __init__(self, state_uuid: str = str(uuid.uuid4()),
         if not conditions:
             conditions = {}
 
+        # the time the state was delayed due to a pause action
+        self.pause_time = 0
```
Comment (simohlsen) [inline server/execution/entities/stategraphs/patientstate.py:21]: If this is a default/none-value, change to `-1`

**Review by simohlsen — CHANGES_REQUESTED** [review]

File: server/execution/entities/execution.py (hunk 1 above)
Comment (YIllmann) [reply server/execution/entities/execution.py:36]: Noteable for Monday Meeting. I would not decide that in this PR.

**Review by simohlsen — APPROVED** [review]

**Pull request merged**